### PR TITLE
Add type alb

### PR DESCRIPTION
**Pull request by kief**
This adds basic support for alb, as per #203. Allows you to check that it exists, and get the dns name.

### CODE REVIEW

File: lib/awspec/helper/finder/alb.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Awspec::Helper
+  module Finder
+    module Alb
+      def find_alb(id)
+        res = elbv2_client.describe_load_balancers({ names: [id] })
+        res.load_balancers.first
```
Comment (k1LoW) [inline lib/awspec/helper/finder/alb.rb]: use `Array#single_resource`

like this

```ruby
res.load_balancers.single_resource(id)
```

File: lib/awspec/type/alb.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  class Alb < Base
+    def resource_via_client
+      @resource_via_client ||= find_alb(@display_name)
+    end
+
+    def id
+      @id = resource_via_client.load_balancer_name if resource_via_client
+    end
+
+    def dns_name
```
Comment (k1LoW) [inline lib/awspec/type/alb.rb]: unneccesary method
(see https://github.com/kief/awspec/blob/44c775f4c76c79856dd3866fae79f0cfd9f201b2/lib/awspec/type/base.rb#L58 )

File: lib/awspec/generator/doc/type/alb.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module Awspec::Generator
+  module Doc
+    module Type
+      class Alb < Base
+        def initialize
+          super
+          @type_name = 'Alb'
```
Comment (k1LoW) [inline lib/awspec/generator/doc/type/alb.rb]: Please `s/Alb/ALB/`

**Comment by k1LoW** [comment]
Hi @kief !  Thank you for your great commit!!
Please check my comment!

Best Regards

File: lib/awspec/type/alb.rb (hunk 2 above)
Comment (kief) [inline lib/awspec/type/alb.rb]: When I remove `resource_via_client` I get an exception. This method is actually defined in the base class, so `method_missing` doesn't get called.

```
  1) alb 'my-alb' should exist
     Failure/Error: raise 'this method must be override!'

     RuntimeError:
       this method must be override!
     # ./lib/awspec/type/base.rb:35:in `resource_via_client'
     # ./lib/awspec/type/alb.rb:8:in `id'
     # ./lib/awspec/type/base.rb:27:in `exists?'
     # ./spec/type/alb_spec.rb:5:in `block (2 levels) in <top (required)>'

```

Comment (k1LoW) [inline lib/awspec/type/alb.rb]: Hi @kief .
remove `dns_name` (not `resource_via_client`)

**Comment by k1LoW** [comment]
Released as v0.63.0.

Nice work! @kief

**Pull request merged**